rotator_pxi/Dockerfile

```diff
@@ -0,0 +1,115 @@
+
+ARG IPIPE_PATCH_FILE=ipipe-core-3.18.20-x86-9.patch
+
+ARG ETHERCAT_VERSION=ethercat-hg
+ARG ETHERCAT_FILE=${ETHERCAT_VERSION}.tar.gz
+ARG ETHERCAT_SOURCE_DIR_HOST=libs/ethercat
+ARG ETHERCAT_SOURCE=/home/ethercat
+
+ARG HDF5_VERSION=hdf5-1.12.0
+ARG HDF5_FILE=${HDF5_VERSION}.tar.gz
+ARG HDF5_SOURCE_DIR_HOST=libs/hdf5
+ARG HDF5_SOURCE_DIR=/home/hdf5
+
+USER root
+
+# Install the required packages
+RUN yum update -y \
+    && yum clean all \
+    && yum groupinstall -y "Development tools" \
+    && yum install -y ncurses-devel hmaccalc zlib-devel binutils-devel \
+    elfutils-libelf-devel wget bc xfsdump nfs-utils grub2 cmake glib-devel \
+    glib2-devel.x86_64 libtool autoconf automake python-devel \
+    && yum clean all \
+    && rm -rf /var/cache/yum
+
+# Install the Linux kernel
+COPY ${LINUX_KERNEL_SOURCE_DIR_HOST}/${LINUX_KERNEL_FILE} /home
+RUN mkdir ${LINUX_SOURCE_DIR} \
+    && cd ${LINUX_SOURCE_DIR} \
+    && mv /home/${LINUX_KERNEL_FILE} . \
+    && tar -xzf ${LINUX_KERNEL_FILE} \
+    && cd ${LINUX_KERNEL_VERSION} \
+    && make defconfig \
+    && make \
+    && make modules_install \
+    && make install \
+    && rm ../${LINUX_KERNEL_FILE}
+
+# Install the Xenomai dual kernel
+COPY ${XENOMAI_SOURCE_DIR_HOST}/${XENOMAI_FILE} /home
+COPY ${XENOMAI_SOURCE_DIR_HOST}/${IPIPE_PATCH_FILE} /home
+RUN mkdir ${XENOMAI_SOURCE_DIR} \
+    && cd ${XENOMAI_SOURCE_DIR} \
+    && mv /home/${XENOMAI_FILE} . \
+    && mv /home/${IPIPE_PATCH_FILE} . \
+    && tar -xf ${XENOMAI_FILE} \
+    && cd ${LINUX_SOURCE_DIR}/${LINUX_KERNEL_VERSION} \
+    && ${XENOMAI_SOURCE_DIR}/${XENOMAI_VERSION}/scripts/prepare-kernel.sh \
+    --arch=x86_64 --ipipe=${XENOMAI_SOURCE_DIR}/${IPIPE_PATCH_FILE} \
+    && rm ${XENOMAI_SOURCE_DIR}/${XENOMAI_FILE}
+
+# Install the Xenomai libraries and tools
+RUN mkdir ${XENOMAI_BUILD_DIR} \
+    && cd ${XENOMAI_BUILD_DIR} \
+    && ${XENOMAI_SOURCE_DIR}/${XENOMAI_VERSION}/configure \
+    && make install
+
+# Install the EtherCAT
+COPY ${ETHERCAT_SOURCE_DIR_HOST}/${ETHERCAT_FILE} /home
+RUN mkdir ${ETHERCAT_SOURCE} \
+    && cd ${ETHERCAT_SOURCE} \
+    && mv /home/${ETHERCAT_FILE} . \
+    && tar -xzf ${ETHERCAT_FILE} \
+    && cd ${ETHERCAT_VERSION} \
+    && ./bootstrap \
+    && ./configure --enable-8139too=no --enable-generic=no \
+    --with-xenomai-dir=/usr/xenomai --enable-cycles \
+    --with-linux-dir=${LINUX_SOURCE_DIR}/${LINUX_KERNEL_VERSION} \
+    && make \
+    && make modules \
+    && make install \
+    && make modules install \
+    && rm ../${ETHERCAT_FILE}
+
+# Install the HDF5
+COPY ${HDF5_SOURCE_DIR_HOST}/${HDF5_FILE} /home
+RUN mkdir ${HDF5_SOURCE_DIR} \
+    && cd ${HDF5_SOURCE_DIR} \
+    && mv /home/${HDF5_FILE} . \
+    && tar -xzf ${HDF5_FILE} \
+    && cd ${HDF5_VERSION} \
+    && autoreconf --install \
+    && ./configure --prefix=/usr/local/hdf5 \
+    && make \
+    && make install \
+    && rm ../${HDF5_FILE}
+
+# Add LSST user
+RUN useradd -ms /bin/bash ${LSST_USER}
```

Comment:
Move this before calling RUN..the trick is if you change thinks to install etc., you will need extra new commit for this command. If you do it before adding new sources,.. it will be taken from repositiory.

Reply:
I am not sure I understand here. The developer can use the `USER` to install the software as `root`. In addition, I do not think there is the problem to add the new user just before the end of `Dockerfile`.

Reply:
It isn't. But you will save one Docker commit if you move RUN useradd -ms.. up just before any other COPY/RUN command, and you later modify something

Reply:
@pkubanek please clarify where you are asking that this be moved. (It may be obvious to @teweitsai but it is not to me).

Reply:
I understand your meaning now. I think it is fine here because the docker has extend the number of layers from 42 to 127 now. Therefore, I think the readability will be better be doing this. Thanks!